### ISSUE TITLE
support form submissions

### ISSUE THREAD

**Issue by techieshark** [MeltanoLabs/tap-hubspot#78]
This would be useful:
https://legacydocs.hubspot.com/docs/methods/forms/get-submissions-for-a-form

> Example GET URL:
> `https://api.hubapi.com/form-integrations/v1/submissions/forms/583d14ff-91af-403d-b94a-c8b5d954da6f`
> 
> Example response:
>```
> {
>  "results": [
>    {
>      "submittedAt": 1547130102576,
>      "values": [
>        {
>          "name": "email",
>          "value": "example@example.com"
>        },
>        {
>          "name": "company",
>          "value": "Example Co"
>        },
>        {
>          "name": "phone",
>          "value": "123 555 4567"
>        }
>      ],
>      "pageUrl": "https://www.example.com/form"
>    },
>```

See also:
* Airbyte implementation https://github.com/airbytehq/airbyte/blob/fbc63ee0616250109bc643127090f689d41e17f6/airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py#L1769 
* gitlab/hotglue implementation: https://gitlab.com/hotglue/tap-hubspot-beta/-/blob/master/tap_hubspot_beta/streams.py?ref_type=heads#L533

Note:
* The Hubspot API is AFAIK somewhat strange in that there are two places where (partial) info about a form submission can be found. First, the form submission API; that gives you things like the email address and other form values. Secondly, the Hubspot Contact API - each contact will have a nested object that has info about form submissions, and this is where one can identify the unique form submission ID (but it doesn't give you the values). So theoretically to get the full picture, as far as I can tell, one would actually need to piece together data from two different APIs. Which is... crazy and annoying, so maybe that should be considered in a separate task at some future point.

**Comment by edgarrmondragon** [comment]
PRs welcome!

**Comment by ReubenFrankel** [comment]
> PRs welcome!

I may end up adding on behalf of a client - just waiting on them to figure out their app scopes so we can make a successful API call.

**Comment by ReubenFrankel** [comment]
> Also interested in this. They may also be accessible with the CRM API using `https://api.hubapi.com/crm/v3/objects/form_submissions`.

Got access - this is incorrect:

```json
{
  "status": "error",
  "message": "Object type FORM_SUBMISSION is not supported by this endpoint, supported object types [APPOINTMENT, CALL, CAMPAIGN, CART, COMMERCE_PAYMENT, COMMUNICATION, COMPANY, CONTACT, CONTRACT, COURSE, DEAL, DEAL_SPLIT, DISCOUNT, EMAIL, ENGAGEMENT, FEE, FEEDBACK_SUBMISSION, GOAL_TARGET, INVOICE, LEAD, LINE_ITEM, LISTING, MEETING_EVENT, NOTE, ORDER, PARTNER_ACCOUNT, PARTNER_CLIENT, POSTAL_MAIL, PRODUCT, QUOTE, QUOTE_TEMPLATE, SERVICE, SUBSCRIPTION, TASK, TAX, TICKET, USER, PARTNER_SERVICE].",
  "correlationId": "a866038c-48b6-44e0-b544-f5300f0c1c64",
  "category": "VALIDATION_ERROR"
}

```

So we need a new forms stream and form submissions child stream (very similar to Hotglue implementation linked above):

- [`GET https://api.hubapi.com/forms/v2/forms`](https://developers.hubspot.com/docs/reference/api/marketing/forms/v2#get-all-forms-from-an-account)
- [`GET https://api.hubapi.com/form-integrations/v1/submissions/forms/{form_id}`](https://developers.hubspot.com/docs/reference/api/marketing/forms/v1#get-submissions-for-a-form)
